### PR TITLE
Remove usages of range that dont describe linear values

### PR DESCRIPTION
**Pull request by rukai**
Fixes https://github.com/gfx-rs/gfx/issues/2571
PR checklist:
- [x] `make` succeeds (on *nix)
- [x] `make reftests` succeeds
- [x] tested examples with the following backends:
   - linux opengl + vulkan

~~- [ ] `rustfmt` run on changed code~~

I haven't fully checked that this covers every case, I just went through the examples and changed every case that pops up there. Can thoroughly replace everything in this PR or a followup.

### CODE REVIEW

File: examples/colour-uniform/main.rs [hunk 1]
```diff
@@ -804,11 +805,12 @@ impl<B: Backend> RenderPassState<B> {
             };
 
             let dependency = pass::SubpassDependency {
-                passes: pass::SubpassRef::External .. pass::SubpassRef::Pass(0),
-                stages: PipelineStage::COLOR_ATTACHMENT_OUTPUT
-                    .. PipelineStage::COLOR_ATTACHMENT_OUTPUT,
-                accesses: i::Access::empty()
-                    .. (i::Access::COLOR_ATTACHMENT_READ | i::Access::COLOR_ATTACHMENT_WRITE),
+                src_subpass: pass::SubpassRef::External,
```
Comment (kvark) [inline examples/colour-uniform/main.rs]: My biggest concern here is duplication of code. Would you be open to a new struct that encloses the duplicated types? Something like

```
struct DependencySide {
  subpass: SubRef,
  stage: PipelineStage,
  access: image::Access,
}
struct SubpassDependency {
   src: DependencySide,
   dst: DependencySide,
}
```

File: examples/colour-uniform/main.rs [hunk 2]
```diff
@@ -1228,15 +1230,18 @@ impl<B: Backend> ImageState<B> {
             cmd_buffer.begin();
 
             let image_barrier = m::Barrier::Image {
-                states: (i::Access::empty(), i::Layout::Undefined)
-                    .. (i::Access::TRANSFER_WRITE, i::Layout::TransferDstOptimal),
+                src_access: i::Access::empty(),
```
Comment (kvark) [inline examples/colour-uniform/main.rs]: we already have a nice `type State = (Access, Layout)` for images, let's use it here

File: examples/colour-uniform/main.rs (hunk 1 above)
Comment (rukai) [reply examples/colour-uniform/main.rs]: Duplication of user code? Or duplication of gfx-hal code?

What are your thoughts on keeping the public API with `src_`/`dest_` fields and then using the DependencySide internally.

Comment (kvark) [reply examples/colour-uniform/main.rs]: duplication on the API side is what I mean

**Comment by rukai** [comment]
I've lost interest in this pr and can't see myself working on it again. Sorry to close it after review had started :/

Still interested in other gfx/wgpu work in the future but I'm fairly busy irl at the moment.

**Pull request closed without merging**